src/Illuminate/Support/Collection.php

```diff
@@ -54,6 +54,21 @@ public function all()
     }
 
     /**
+     * Get the average value of a given key.
+     *
+     * @param  string|null  $key
+     * @return mixed
+     */
+    public function avg($key = null)
+    {
+        if ($this->count() == 0) {
```

Comment:
@busayo I think that @lucasmichot don't want to compare `$count` and `$this->count()`. He wants to assign to `$count` returned value of `$this->count()`.

If `$this->count() >= 0` then body of if will be executed. Way offered by @lucasmichot  prevent us from counting elements of collection twice

If you want it can be
```php
    if (($count = $this->count()) != 0) {
        return $this->sum($key) / $count;
    }
```